### DISCUSSION TITLE
How to deal with image widget that may not have an image to display

### DISCUSSION THREAD

**Discussion by Nemo157** [elkowar/eww#933] [Q&A]
I'm writing a little music widget for my taskbar, if there's no music status then it gets hidden, but the logs are spammed with failure to access the image file:

```
 2023-09-28T17:11:44.996Z ERROR eww::error_handling_ctx > Error while updating UI after state change

Caused by:
    Failed to open file “”: No such file or directory
```

trying both leaving a json `null` value or an empty string fails. My current plan is to find some image to put there as a default, but since it's not even shown it feels like this shouldn't be necessary. (Unless there's something simple I'm missing that allows for actually dynamically adding the widget instead of just making it invisible).

```
(defwidget music []
  (box :orientation "v" :class "music" :space-evenly false :visible {music != "{}"}
    ... other sub widgets ...
    (image :class "albumart" :image-width 50 :path {music?.albumart ?: ""})
    ... other sub widgets ...
  )
)
```

**Comment by V1v4ldi** [comment]
have the same problem, my workaround is just shove it with empty image. cause if i see it correctly, the image in your code is hidden when there's no music right?
